### ISSUE TITLE
History-graph: timeline: glitches if a state=unknown

### ISSUE THREAD

**Issue by ildar170975** [home-assistant/frontend#18800]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

(will move this to Core if needed)

Tested 2 `person` entities, one of them was `unknown` (some issues with a source device_tracker).
Have a look at this History:

![изображение](https://github.com/home-assistant/frontend/assets/71872483/8d239b44-85cf-4907-ba79-e2c5fed12d24)

The upper entity seems to be in some zone.
Wrong.
It is `unknown`:

![изображение](https://github.com/home-assistant/frontend/assets/71872483/c6bf7061-6922-4c44-8fd7-8d7329487687)

Let's change a left boundary in the History.
Now the 1st entity is shown as `unknown`:

![изображение](https://github.com/home-assistant/frontend/assets/71872483/55773e65-ded0-4da9-8b35-a33bbe05079d)

Now let's again change a left boundary in the History.
Now the 1st entity is NOT shown at all, and the 2nd entity is shown w/o it's "Home" timeline:

![изображение](https://github.com/home-assistant/frontend/assets/71872483/6e12161d-8b00-4897-b036-45435084949a)

### Describe the behavior you expected

CORRECT GRAPHS

### Steps to reproduce the issue

as above

### What version of Home Assistant Core has the issue?

2023.11.2

### In which browser are you experiencing the issue with?

Chrome 119.0.6045.160 

### Which operating system are you using to run this browser?

Win10

**Comment by karwosts** [comment]
I did a little bit of testing but can't reproduce anything like this. 🤷 

I see your graph does sometimes say unknown; what times does the graph think the entity changed states, and what time do you actually have state changes in your states table?

**Comment by ildar170975** [comment]
> Are these graphs reproducible if you refresh the page and go to the same time range?

I reproduced this glitch on another browser - FF:
-- open HA in FF
-- go to History
-- select these 2 entities
-- select these time ranges - in each case same picture.

> I see your graph does sometimes say unknown; what times does the graph think the entity changed states, and what time do you actually have state changes in your states table?

First I started explaining why a curr. situation in Russia caused problem in location tracking -> and this caused issues with `person` entities, but this could be not interesting for foreign people & too long to read & emotional.

In short: due to some reasons a `device_tracker` may become `unavailable`.
So, a corr. `person` entity becomes `unknown`.
This may be simulated in Dev tools:

![изображение](https://github.com/home-assistant/frontend/assets/71872483/f2cbf089-2f23-4a46-9bc3-3cd8372e83ba)

In that particular case there were issues with 2 mobile phones (mine & Dad's) with [Traccar integration](https://www.home-assistant.io/integrations/traccar/).
So, corr. `person` entities became `unknown`...

**Comment by karwosts** [comment]
Yes I did simulate it in devtools, I forced my person to `unknown`, and a few minutes later I set it back to `home`, and viewing this in the graph looked completely as expected, across multiple various time ranges. 

So was just trying to guess what might be different in your case. I was curious about what the actual states in your history database say, compared to what the graph shows.

**Comment by ildar170975** [comment]
First I was going to update my post. Will post it here.

Here in that incident there were probably another reason.
GPS & GLONASS location here sometimes is very unstable. Mobile devices report wrong location data. Sometimes phones jump to very far points for some period.
As a part of processing these cases, particularly in [Traccar integration](https://www.home-assistant.io/integrations/traccar/) I had to set a ["max_accuracy" parameter](https://www.home-assistant.io/integrations/traccar/#max_accuracy). It allows to filter locations with a low accuracy - so a corr. `device_tracker` entity will not have these `latitude` & `longitude`.
But this may cause problems: if HA was rebooted after a moment when a location stopped updating - then at HA startup a corr. `device_tracker` may not get `latitude` & `longitude` at all.
And a `person` will be again `unknown` - until the mobile device manages to get a "good" location data and these data will be used in HA.

I may only provide HA logs (no related errors), history & screenshots. Unfortunately, not qualified enough to extract data from DB.

**Comment by ildar170975** [comment]
Will close it since cannot reproduce.
